tox.ini

```diff
@@ -28,9 +28,11 @@ allowlist_externals =
     find
     pip-compile
     xargs
-commands = 
-; uses 'bash -c' because piping didn't work in regular tox commands
-  bash -c 'find . -type f -name "requirements*.in" | xargs --replace=\{\} pip-compile -U \{\}'
+commands =
+    ; we must preserve the order of compilation, since each *.in file depends on some *.txt file.
+    ; For example, requirements-unit.in depends on requirements.txt and we must compile first
+    ; requirements.txt to ensure that requirements-unit.txt get the same dependency as the requirements.txt
+    bash -c 'for pattern in "requirements.in" "requirements-fmt.in" "requirements*.in"; do find . -type f -name "$pattern" -exec bash -c "cd \$(dirname "{}") && pip-compile --resolver=backtracking \$(basename "{}")" \;; done'
```

Comment:
We can remove `--resolver=backtracking` to simplify, since it's now the default in the new pip-tools.

Also, why not include the upgrade flag there (i.e. `-U`)?

Reply:
BTW, this is the order of files returned by `find` before:
```
./requirements-fmt.in
./requirements.in
./requirements-unit.in
./requirements-integration.in
./requirements-lint.in
```
I get that's implicit, but I don't necessarily understand why that's harder to understand than the bash script above😇

Reply:
I think this guarantees that the first item on the list is always requirements.in, which prevents issues with other .in files that depend on the .txt that's generated from it.

Reply:
> We can remove `--resolver=backtracking` to simplify, since it's now the default in the new pip-tools.
> 
> Also, why not include the upgrade flag there (i.e. `-U`)?

`-U` will try to upgrade ALL packages, which I would not recommend every time as upgrading should be handled by `renovate`. Ideally we would not want to upgrade every time. I see this environment as something to use only when adding or removing a dependency to any of the .in files, if we'd want to explicitly upgrade one or more packages, I think pip-compile should be used targeting the package(s) to upgrade.